prowler/providers/aws/services/networkfirewall/networkfirewall_service.py

```diff
@@ -27,13 +27,14 @@ def _list_firewalls(self, regional_client):
                             network_firewall["FirewallArn"], self.audit_resources
                         )
                     ):
-                        self.network_firewalls.append(
+                        self.network_firewalls[network_firewall.get("FirewallArn")] = (
                             Firewall(
                                 arn=network_firewall.get("FirewallArn"),
```

Comment:
Please could you remove the ARN from the model since that information is already in the key